feat(http): Support for `Retry-After` header (#25859)
renovatebot/renovate: docs/usage/configuration-options.md

}
```

### maxRetryAfter

> By default, Renovate will retry after any `Retry-After` header value, up to a maximum of 60 seconds.

Reviewer: ```suggestion
At times, remote hosts will return a 4xx response with a `Retry-After` header value, indicating that Renovate has been rate limited but may retry after a certain waiting period.
By default, Renovate will retry after any `Retry-After` header value, up to a maximum of 60 seconds.
If the `Retry-After` value is higher than 60 seconds, Renovate will instead abort the job.
```

Reviewer 2: ```suggestion
A remote host may return a `4xx` response with a `Retry-After` header value, which indicates that Renovate has been rate-limited.
Renovate may try to contact the host again after waiting a certain time, that's set by the host.
By default, Renovate tries again after the `Retry-After` header value has passed, up to a maximum of 60 seconds.
If the `Retry-After` value is more than 60 seconds, Renovate will abort the job instead of waiting.
```

How about something like this?

}
```

### maxRetryAfter

A remote host may return a `4xx` response with a `Retry-After` header value, which indicates that Renovate has been rate-limited.
Renovate may try to contact the host again after waiting a certain time, that's set by the host.
By default, Renovate tries again after the `Retry-After` header value has passed, up to a maximum of 60 seconds.

> If the `Retry-After` value is more than 60 seconds, Renovate will abort the job instead of waiting.

Reviewer 2: ```suggestion
If the `Retry-After` value is more than 60 seconds, Renovate will abort the request instead of waiting.
```

> You can configure a different maximum value using `maxRetryAfter`:

Reviewer 2: The docs say that by default, Renovate will abort the run after waiting `maxRetryAfter=60`.

What happens when a user sets `maxRetryAfter=120`? Does Renovate still abort after 60 seconds?

Maybe the abort behavior depends on self-hosting/hosted app? Where the self-hosted version can stretch the waiting time a lot more, and the hosted app always aborts after 60 seconds?

Reviewer: Should be abort the request, not abort the job. The job _might_ be aborted, but that's a different logic

> You can configure a different maximum value using `maxRetryAfter`:

Reviewer 3: ```suggestion
You can configure a different maximum value in seconds using `maxRetryAfter`:
```
